src/Microsoft.ML.Transforms/CategoricalTransform.cs

```diff
@@ -301,4 +312,116 @@ public static CommonOutputs.TransformOutput KeyToText(IHostEnvironment env, KeyT
+        private const int DefMax = TermEstimator.Defaults.MaxNumTerms;
+        private const OneHotOutputKind DefOut = (OneHotOutputKind)CategoricalEstimator.Defaults.OutKind;
+
+        private struct Config
+        {
+            public readonly KeyValueOrder Order;
+            public readonly int Max;
+            public readonly OneHotOutputKind OutputKind;
+
+            public Config(OneHotOutputKind outputKind, KeyValueOrder order, int max)
+            {
+                OutputKind = outputKind;
+                Order = order;
+                Max = max;
+            }
+        }
+
+        private interface ICategoricalCol
+        {
+            PipelineColumn Input { get; }
+            Config Config { get; }
+        }
+
+        private sealed class ImplScalar<T> : Vector<float>, ICategoricalCol
+        {
+            public PipelineColumn Input { get; }
+            public Config Config { get; }
+            public ImplScalar(PipelineColumn input, Config config) : base(Rec.Inst, input)
+            {
+                Input = input;
+                Config = config;
+            }
+        }
+
+        private sealed class ImplVector<T> : Vector<float>, ICategoricalCol
+        {
+            public PipelineColumn Input { get; }
+            public Config Config { get; }
+            public ImplVector(PipelineColumn input, Config config) : base(Rec.Inst, input)
+            {
+                Input = input;
+                Config = config;
+            }
+        }
+
+        private sealed class Rec : EstimatorReconciler
+        {
+            public static readonly Rec Inst = new Rec();
+
+            public override IEstimator<ITransformer> Reconcile(IHostEnvironment env, PipelineColumn[] toOutput,
+                IReadOnlyDictionary<PipelineColumn, string> inputNames, IReadOnlyDictionary<PipelineColumn, string> outputNames, IReadOnlyCollection<string> usedNames)
+            {
+                var infos = new CategoricalEstimator.ColumnInfo[toOutput.Length];
+                for (int i = 0; i < toOutput.Length; ++i)
+                {
+                    var tcol = (ICategoricalCol)toOutput[i];
+                    infos[i] = new CategoricalEstimator.ColumnInfo(inputNames[tcol.Input], outputNames[toOutput[i]], (CategoricalTransform.OutputKind)tcol.Config.OutputKind,
+                        tcol.Config.Max, (TermTransform.SortOrder)tcol.Config.Order);
+                }
+                return new CategoricalEstimator(env, infos);
+            }
+        }
+
+        /// <summary>
+        /// Converts the categorical value into an indicator array by building a dictionary of categories based on the data and using the id in the dictionary as the index in the array
+        /// </summary>
+        /// <param name="input">Incoming data.</param>
+        /// <param name="outputKind">Specify output type of indicator array: Multiarray, array or binary encoded data.</param>
```

Comment:
>Multiarray, array [](start = 77, length = 17)

In the scalar case there is no distinction between these two, so presenting this as an option is at least confusing. Is it worthwhile to clarify this?